guardrails/cli/hub/console.py

```diff
@@ -0,0 +1,3 @@
+from rich.console import Console
+
+console = Console()
```

Comment:
@CalebCourier to clarify, this doesn't replace our existing logger but adds another console object for the top level CLI for us to render complex objects. We can independently continue using the current logger to log statements.

I looked into this, unfortunately a lot of the CLI Renderables that `rich` supports aren't supported by the RichHandler. Specifically for our case, the status updates won't be supported with a rich handler. I can add back the original log statements with a `DEBUG` mode.

Reply:
By "replace" I meant in practice, as in where the logs are actually going.  This is probably something we can reconcile later, but since the `console.print` statements bypass the logging system instead of working with it, any internal usage of a logger from pip, or even the modules being installed, will interrupt the console output.

Example:
<img width="913" alt="Screenshot 2024-04-05 at 8 16 33 AM" src="https://github.com/guardrails-ai/guardrails/assets/13314870/080a0ded-e73b-46ee-86c3-24d1a14f595e">


It also means that if we wanted to add a logging handler to collect and send/process logs elsewhere (like for telemetry), any logs sent to `console` instead of a the logger wouldn't show up.  This may be fine since the logs sent to console are more aesthetic vs technical, but it's still a potential limitation.

In general, I think these are all things that can be addressed later but I do think we should at least acknowledge them before committing to this pattern.